src/unix/core.c

```diff
@@ -1284,6 +1288,70 @@ int uv_translate_sys_error(int sys_errno) {
 }
 
 
+int uv_os_environ(uv_env_item_t** envitems, int* count) {
+  int i, j, cnt;
+  uv_env_item_t* envitem;
+
+  *envitems = NULL;
+  *count = 0;
+
+  for (i = 0; environ[i]; i++);
+
+  *envitems = uv__calloc(i, sizeof(**envitems));
```

Comment:
Why using `calloc` instead of `malloc`?

Reply:
because it's made for arrays and because it zeroes out the memory so you end up with a nice list of all nulls :-)

Reply:
Yeah but, isn't the whole array (except if no `=` is found) being filled anyway? If that's the case, wouldn't it be more optimal without zeroing the memory?

Reply:
There is also a case I forgot to handle: if the environment is mutated while we work on it. We may end up with a shorter version. I think it’s safer to zero upfront.

Reply:
Yeah, I guess another thread could unset some variables. :+1: